src/DotNetProjectFile.Analyzers/IO/IOFile.cs

```diff
@@ -70,6 +70,22 @@ public string NameWithoutExtension
     /// <inheritdoc cref="FileInfo.OpenRead()" />
     public FileStream OpenRead() => File().OpenRead();
 
+    /// <summary>Tries <see cref="OpenRead()"/>.</summary>
+    /// <remarks>
+    /// Returns <see cref="Stream.Null"/> if reading fails.
+    /// </remarks>
+    public Stream TryOpenRead()
+    {
+        try
+        {
+            return OpenRead();
```

Comment:
What is the difference between using OpenRead() and File().OpenRead()?

Reply:
`IOFile` is wrapper around `FileInfo` (not totally, but to a high extend). `IOFile.OpenRead()` exposes `FileInfo.OpenRead()`, by creating a `FileInfo` on the fly, and calling it's `.OpenRead()`. So method body if we would have replaced `OpenRead()` with `File().OpenRead()` we would had have the same behavior, but with code duplication.